lib/zeromq/majordomo/asynchronous_arbiter.rb

```diff
@@ -0,0 +1,44 @@
+require 'ffi-rzmq'
+
+module Zeromq
+  module Majordomo
+    class AsynchronousArbiter
+      INFINITE = -1
+
+      def initialize(address, context, timeout_in_sec, md_service)
+        @address = address
+        @context = context
+        @timeout = timeout_in_sec.to_i * 1000 #ms
+        @md_service = md_service
+      end
+
+      def publish(method, params)
+        sock = connect
+        assert_zmq_ok(sock.send_strings(
+          ['', 'MDPC01', @md_service, method.to_s, Marshal.dump(params)]
+        ))
+      ensure
+        disconnect(sock)
```

Comment:
There's actually a subtle bug here. If an exception is raised in the `connect` method, `sock` won't exist. This should probably be:

```ruby
disconnect(sock) if sock
```